src/Streamly/Prelude.hs

```diff
@@ -332,11 +335,26 @@ head m =
         yield a _ = return (Just a)
     in (runStream (toStream m)) Nothing stop yield
 
+-- | Extract all but the first element of the stream, if any.
+tail :: (Streaming t, Monad m) => t m a -> m (Maybe (t m a))
+tail m =
+    let stop             = return Nothing
+        yield _ Nothing  = return $ Just nil
```

Comment:
It should be `return Nothing`, we do not want to return an empty stream as tail when there is no tail. I see you have missed the test for tail.